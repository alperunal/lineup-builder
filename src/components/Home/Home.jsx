import React from 'react';
import { Link } from 'react-router-dom';
import { injectIntl, FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import { Container, Card } from 'components/UI';

import heroImgSrc from 'assets/images/hero.png';
import footballImgSrc from 'assets/images/dragndrop.png';
import footballFieldImgSrc from 'assets/images/fieldtype.png';
import footballPlayersImgSrc from 'assets/images/formation.png';

import './Home.module.scss';

const Home = ({ intl }) => (
    <div className="home">
        <div className="hero">
            <Container>
                <div className="hero__items">
                    <div className="hero__item">
                        <h1>
                            <FormattedMessage
                                id="home.slogan"
                                defaultMessage="Build Your Lineup"
                            />
                        </h1>
                        <p className="hero__description">
                            <FormattedMessage
                                id="home.description"
                                defaultMessage="Create your own Football formation using Voety Lineup Builder.
                                Customize and share it using image download, save and load features."
                            />
                        </p>
                        <Link to="/lineup" className="hero__button">
                            <FormattedMessage
                                id="home.buildnow"
                                defaultMessage="Build Now"
                            />
                        </Link>
                    </div>
                    <div className="hero__item hero__item--right">
                        <img src={heroImgSrc} alt="lineup" />
                    </div>
                </div>
            </Container>
        </div>
        <div className="home__start">
            <Container>
                <h1>
                    <FormattedMessage
                        id="home.howtoplay"
                        defaultMessage="How To Play"
                    />
                </h1>
                <div className="home__cards">
                    <Card
                        customClass="home__card"
                        imageSrc={footballPlayersImgSrc}
                        imageAlt="Select Formation"
                        title={intl.formatMessage({ id: 'home.card1.title', defaultMessage: 'Select Formation' })}
                        description={intl.formatMessage({ id: 'home.card1.description', defaultMessage: 'Select one of the ready to use formations' })}
                    />
                    <Card
                        customClass="home__card"
                        imageSrc={footballFieldImgSrc}
                        imageAlt="Select Field"
                        title={intl.formatMessage({ id: 'home.card2.title', defaultMessage: 'Select Field' })}
                        description={intl.formatMessage({ id: 'home.card2.description', defaultMessage: 'Select a field type' })}
                    />
                    <Card
                        customClass="home__card"
                        imageSrc={footballImgSrc}
                        imageAlt="Drag'n Drop"
                        title={intl.formatMessage({ id: 'home.card3.title', defaultMessage: 'Drag\'n Drop' })}
                        description={intl.formatMessage({ id: 'home.card3.description', defaultMessage: 'Modify your formation via drag and drop' })}
                    />
                </div>
            </Container>
        </div>
    </div>
);

Home.propTypes = {
    intl: PropTypes.shape({
        formatMessage: PropTypes.func.isRequired,
    }).isRequired,
};

export default injectIntl(Home);
