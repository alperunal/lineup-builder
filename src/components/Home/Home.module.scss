.home {
    .hero {
        background-image: radial-gradient(70% 70% at 50% 100%,#e8e8e8 0,#fafafa 100%);
        padding: 150px 0;

        &__items {
            display: flex;
            flex-direction: column;

            @media(min-width: 1000px) {
                flex-direction: row;
            }
        }
    
        &__item {

            @media(min-width: 1000px) {
                width: 50%;
            }
            
            &--right {
                display: none;
                justify-content: flex-end;

                @media(min-width: 1000px) {
                    display: flex;
                }
            }
        }

        &__description {
            font-size:18px;
            letter-spacing:-0.01em;
            line-height: 1.7;
            font-weight: 400;
        }

        &__button {
            animation: btn-gradient-animation 3s ease-in-out infinite;
            border: none;
            border-radius: 5px;
            font-family: 'Roboto', sans-serif;
            font-weight: 500;
            font-size: 18px;
            padding: 15px 68px;
            margin-top: 50px;
            color: #fff !important;
            text-decoration: none !important;
            cursor: pointer;
            outline: 0;
            text-transform: uppercase;
            vertical-align: middle;
            -webkit-tap-highlight-color: transparent;
            text-align: center;
            letter-spacing: 3px;
            display: inline-block;
            background: #F17C35;
            background: linear-gradient(45deg, #F17C35 0%, #E2336E 44%, #E2336E 56%, #F17C35 100%);
            background-size: 150% 150%;
            transition: background-position 0.7s ease-in-out;
        }
    }

    &__start {
        padding: 125px 0;
    }

    &__cards {
        display: flex;
        flex-direction: column;
        align-items: center;

        @media(min-width: 1000px) {
            flex-direction: row;
            justify-content: center;
        }
    }

    &__card {
        width: 250px;
        margin: 25px;

        @media(min-width: 1000px) {
            width: 25%;
        }
    }

    
    h1 {
        font-size: 3em;
    }

    h2 {
        font-size: 2em;
        color: #666666;
    }
}