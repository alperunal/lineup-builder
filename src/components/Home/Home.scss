@import 'styles/_variables.scss';

.home {
  text-align: center;

  @media (min-width: $breakpoint--lg) {
    text-align: left;
  }

  .hero {
    background: var(--color-hero-background);
    padding: 150px 0;

    &__items {
      display: flex;
      flex-direction: column;

      @media (min-width: $breakpoint--lg) {
        flex-direction: row;
      }
    }

    &__item {
      @media (min-width: $breakpoint--lg) {
        width: 50%;
      }

      &--right {
        display: none;
        justify-content: flex-end;

        @media (min-width: $breakpoint--lg) {
          display: flex;
        }
      }
    }

    &__description {
      font-size: 18px;
      letter-spacing: -0.01em;
      line-height: 1.7;
      font-weight: 400;
    }

    &__button {
      animation: btn-gradient-animation 3s ease-in-out infinite;
      border: none;
      border-radius: 5px;
      font-weight: bold;
      font-size: 18px;
      padding: 15px 68px;
      margin-top: 50px;
      color: #fff !important;
      text-decoration: none !important;
      cursor: pointer;
      outline: 0;
      text-transform: uppercase;
      vertical-align: middle;
      -webkit-tap-highlight-color: transparent;
      text-align: center;
      letter-spacing: 3px;
      display: inline-block;
      background: #f17c35;
      background: linear-gradient(
        45deg,
        #f17c35 0%,
        #e2336e 64%,
        #e2336e 56%,
        #f17c35 100%
      );
      background-size: 150% 150%;
      transition: background-position 0.7s ease-in-out;
    }
  }

  &__start {
    padding: 125px 0;
  }

  &__cards {
    display: flex;
    flex-direction: column;
    align-items: center;

    @media (min-width: $breakpoint--lg) {
      flex-direction: row;
      justify-content: center;
    }
  }

  &__card {
    width: 250px;
    margin: 25px;

    @media (min-width: $breakpoint--lg) {
      width: 25%;
    }
  }

  h1 {
    font-size: 3em;
  }

  h2 {
    font-size: 2em;
    color: #666666;
  }
}
