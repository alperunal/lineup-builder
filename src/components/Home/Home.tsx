import React from 'react';
import { Container, Card } from 'components/UI';
import heroImgSrc from 'assets/images/hero.png';
import './Home.scss';

const Home: React.FC = () => (
    <div className="home">
        <div className="home__hero">
            <Container>
                <div className="home__hero-items">
                    <div className="home__hero-item">
                        <h1>Build Your Line-up</h1>
                    </div>
                    <div className="home__hero-item home__hero-item--right">
                        <img src={heroImgSrc} alt="lineup" />
                    </div>
                </div>
            </Container>
        </div>
        <div className="home__start">
            <Container>
                <h1>Start with National Teams</h1>
                <h2>Select one of the team templates.</h2>
                <div className="cards">
                    <Card
                        customClass="home__card"
                        imageSrc="https://tailwindcss.com/img/card-top.jpg"
                        imageAlt="Netherlands"
                        title="Netherlands"
                        description="2020 Home Example"
                    />
                </div>
            </Container>
        </div>
    </div>
);

export default Home;
