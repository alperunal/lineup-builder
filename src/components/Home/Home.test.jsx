import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

it('should render Footer correctly', () => {
  const wrapper = shallow(<Home />);
  expect(wrapper).toMatchSnapshot();
});
