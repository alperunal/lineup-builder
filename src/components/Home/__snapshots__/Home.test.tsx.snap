// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer correctly 1`] = `ShallowWrapper {}`;
